tests/devices/test_default_qutrit_mixed.py

```diff
@@ -1270,3 +1270,305 @@ def qfunc(x, y):
+                qml.expval(qml.THermitian(non_commuting_obs_two, 0)),
+                qml.expval(qml.GellMann(1, 1)),
+            )
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    state_gammas_and_probs = [
+        [(0, 0, 0), (0, 0, 0)],
+        [(0.1, 0.15, 0.25), (0.1, 0.15, 0.25)],
+        [(1, 0, 1), (1, 0, 0)],
+    ]
+
+    @pytest.mark.parametrize("gammas, probs", state_gammas_and_probs)
+    def test_readout_state(self, nr_wires, gammas, probs):
+        """Tests the state output is not affected by readout error"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            damping_measurement_gammas=gammas,
+            trit_flip_measurement_probs=probs,
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            self.setup_state(nr_wires)
+            return qml.state()
+
+        res = circuit()
+        assert np.allclose(res, self.get_expected_dm(nr_wires))
+
+    @pytest.mark.parametrize("gammas, probs", state_gammas_and_probs)
+    def test_readout_density_matrix(self, nr_wires, gammas, probs):
+        """Tests the density matrix output is not affected by readout error"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            damping_measurement_gammas=gammas,
+            trit_flip_measurement_probs=probs,
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            self.setup_state(nr_wires)
+            return qml.density_matrix(wires=1)
+
+        res = circuit()
+        assert np.allclose(res, self.get_expected_dm(1))
+
+    @pytest.mark.parametrize(
+        "gammas, probs, expected",
+        [
+            ((0, 0, 0), (0, 0, 0), [(np.ones(2) * 2)] * 2),
+            (None, (0, 0, 1), [np.ones(2)] * 2),
+            ((0, 0, 1), None, [np.ones(2)] * 2),
+            (None, (0, 1, 0), [np.zeros(2)] * 2),
+            ((0, 1, 0), None, [np.zeros(2)] * 2),
+        ],
+    )
+    def test_readout_sample(self, nr_wires, gammas, probs, expected):
+        """Tests the sample output with readout error"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            shots=2,
+            wires=nr_wires,
+            damping_measurement_gammas=gammas,
+            trit_flip_measurement_probs=probs,
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.QutritBasisState([2] * nr_wires, wires=range(nr_wires))
+            return qml.sample(wires=[0, 1])
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    @pytest.mark.parametrize(
+        "gammas, probs, expected",
+        [
+            ((0, 0, 0), (0, 0, 0), {"22": 100}),
+            (None, (0, 0, 1), {"11": 100}),
+            ((0, 0, 1), None, {"11": 100}),
+            (None, (0, 1, 0), {"00": 100}),
+            ((0, 1, 0), None, {"00": 100}),
+        ],
+    )
+    def test_readout_counts(self, nr_wires, gammas, probs, expected):
```

Comment:
question: Is this fixed specifically so that the readout results will always be wrong in a predictable way?

Reply:
related question: is there a straightforward way to add tests for when readout error _do_ occur, like with the expectation values?

Reply:
Question 1, yes the probabilities are setup so that it will always be predictable.

Question 2, Yes this could be done either by setting tolerances and a seed (so it is consistent) or by checking that each value is only one of x possible values.